### PR TITLE
Health Expiration Notifications

### PR DESCRIPTION
**Pull request by eanders**
This completes:
https://www.pivotaltracker.com/story/show/169439537
https://www.pivotaltracker.com/story/show/169439579
and the unrelated
https://www.pivotaltracker.com/story/show/169603906

### CODE REVIEW

File: app/views/warehouse_reports/health/expiring_items/_chas.haml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      %thead
+        %tr
+          %th Patient
+          %th Expiration Date
+      %tbody
+        - @report.expired_cha_forms.each do |patient|
+          %tr
+            %td= link_to patient.name, client_health_patient_index_path(patient)
+            %td= patient.recent_cha_form.expires_on || 'Incomplete'
+
```
Comment (awisema) [inline app/views/warehouse_reports/health/expiring_items/_chas.haml]: Should there be a 'No expired or expiring CHAs' etc?

File: app/controllers/warehouse_reports/health/ssm_exports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    def hmis_responses
+      hmis_touch_point_scope.select(
+        hmis_form_t[:id].to_sql,
+        hmis_form_t[:client_id].to_sql,
+        hmis_form_t[:answers].to_sql,
+        hmis_form_t[:collected_at].to_sql,
+        hmis_form_t[:data_source_id].to_sql,
+        hmis_form_t[:assessment_id].to_sql,
+        hmis_form_t[:site_id].to_sql,
+        hmis_form_t[:staff].to_sql,
```
[the remark this answers is not in the document]
Comment (eanders) [reply app/controllers/warehouse_reports/health/ssm_exports_controller.rb]: So, this was necessary if you want to call count on it.  I don't know why, but it was very confused when I was working through some issues.  It's probably unnecessary as written for now, but doesn't hurt.

File: app/controllers/warehouse_reports/health/ssm_exports_controller.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      ).
+        joins(:hmis_assessment, client: :destination_client).
+        order(:client_id, :collected_at)
+    end
+
+    private def patients_by_client_id
+      client_ids = @client_ids.to_a # from HMIS
```
[the remark this answers is not in the document]
Comment (eanders) [reply app/controllers/warehouse_reports/health/ssm_exports_controller.rb]: Yes, this all needs to be cleaned up.  I'm fairly certain this won't run in real-time against production data and will need to be moved off to a model and delayed job.  I'll leave it for now, but cleanup when that happens.

File: app/views/warehouse_reports/health/expiring_items/_chas.haml (hunk 1 above)
Comment (eanders) [reply app/views/warehouse_reports/health/expiring_items/_chas.haml]: I simply didn't show them if there weren't any.  It seemed cleaner, but maybe.

**Pull request merged**